src/libsyntax/parse/parser.rs

```diff
@@ -767,6 +767,9 @@ impl<'a> Parser<'a> {
             err.span_label(self.span, format!("expected identifier, found {}", token_descr));
         } else {
             err.span_label(self.span, "expected identifier");
+            if self.token == token::Comma {
```

Comment:
I'm not sure this is the best thing to do, as is. I would only add this suggestion if the next token is indeed an ident:

```
    if self.token == token::Comma && self.look_ahead(1, |t| *t.is_ident()) {
```

Reply:
I think it will still be useful even if the next token is not an identifier. We should not ever have 2 commas in a row in a struct constructor.
```rust
struct Bar {
    a: u8,
    b: u8
}

let foo = Bar {
    a: 42,
    b: 42,,
};
```
In this example, the user still has to remove the comma to get his code to compile.